src/BenchmarkDotNet/Toolchains/GeneratorBase.cs

```diff
@@ -119,7 +119,16 @@ private ArtifactsPaths GetArtifactsPaths(BuildPartition buildPartition, string r
             string programName = buildPartition.ProgramName;
             string buildArtifactsDirectoryPath = GetBuildArtifactsDirectoryPath(buildPartition, programName);
             string binariesDirectoryPath = GetBinariesDirectoryPath(buildArtifactsDirectoryPath, buildPartition.BuildConfiguration);
-            string executablePath = Path.Combine(binariesDirectoryPath, $"{programName}{GetExecutableExtension()}");
+
+            string executablePath;
```

Comment:
please follow the existing pattern: introduce a new virtual method `GetExecutablePath` with `Path.Combine(binariesDirectoryPath, $"{programName}{GetExecutableExtension()}")` as default implementation and override it in `WasmGenerator` and return `Path.Combine(binariesDirectoryPath, "run-v8.sh")` there